utils/tinymix.c

```diff
@@ -37,38 +37,62 @@
 #define OPTPARSE_IMPLEMENTATION
 #include "optparse.h"
 
-static void tinymix_list_controls(struct mixer *mixer, int print_all);
+static void list_controls(struct mixer *mixer, int print_all);
 
-static void tinymix_detail_control(struct mixer *mixer, const char *control);
+static void print_control_values(struct mixer_ctl *control);
+static void print_control_values_by_name_or_id(struct mixer *mixer, const char *name_or_id);
 
-static int tinymix_set_value(struct mixer *mixer, const char *control,
+static int set_values(struct mixer *mixer, const char *control,
                              char **values, unsigned int num_values);
 
-static void tinymix_print_enum(struct mixer_ctl *ctl);
+static void print_enum(struct mixer_ctl *ctl);
 
 void usage(void)
 {
     printf("usage: tinymix [options] <command>\n");
     printf("options:\n");
-    printf("\t-h, --help        : prints this help message and exits\n");
-    printf("\t-v, --version     : prints this version of tinymix and exits\n");
-    printf("\t-D, --card NUMBER : specifies the card number of the mixer\n");
+    printf("\t-h, --help               : prints this help message and exits\n");
+    printf("\t-v, --version            : prints this version of tinymix and exits\n");
+    printf("\t-D, --card NUMBER        : specifies the card number of the mixer\n");
+    printf("\n");
     printf("commands:\n");
-    printf("\tget NAME|ID       : prints the values of a control\n");
-    printf("\tset NAME|ID VALUE : sets the value of a control\n");
-    printf("\tcontrols          : lists controls of the mixer\n");
-    printf("\tcontents          : lists controls of the mixer and their contents\n");
+    printf("\tget NAME|ID              : prints the values of a control\n");
+    printf("\tset NAME|ID VALUE(S) ... : sets the value of a control\n");
+    printf("\t\tVALUE(S): integers, percents, and relative values\n");
+    printf("\t\t\tIntegers: 0, 100, -100 ...\n");
+    printf("\t\t\tPercents: 0%%, 100%% ...\n");
+    printf("\t\t\tRelative values: 1+, 1-, 1%%+, 2%%+ ...\n");
+    printf("\tcontrols                 : lists controls of the mixer\n");
+    printf("\tcontents                 : lists controls of the mixer and their contents\n");
 }
 
 void version(void)
 {
     printf("tinymix version 2.0 (tinyalsa version %s)\n", TINYALSA_VERSION_STRING);
 }
 
+static int is_command(char *arg) {
+    return strcmp(arg, "get") == 0 || strcmp(arg, "set") == 0 ||
+            strcmp(arg, "controls") == 0 || strcmp(arg, "contents") == 0;
+}
+
+static int find_command_position(int argc, char **argv)
+{
+    int i;
```

Comment:
Why the C89 syntax for the for loop here?

Reply:
I found that most of our code in c language follows the C89 standard. I think it would be better to keep the same style in  this project. What do you think?

Reply:
This got brought up before in a previous issue and I was against being compliant with C89 unless there was a compelling reason to continue using it. I haven't checked in a while but I think most/all the build systems compile with `-std=c99` or `-std=gnu99`.

Reply:
Got it. I think it should not build break while using C99 standard. In pcm_set_config, there are some variables declared at the middle of function. I will modify it.